java/code/src/com/redhat/rhn/domain/server/Pillar.java

```diff
@@ -330,7 +330,7 @@ public boolean equals(Object oIn) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(id, minion, org, group, category, pillar);
+        return Objects.hash(minion, org, group, category);
```

Comment:
I get why `id` must be omitted, but why `pillar` as well? Isn't it in the same boat as `category`?

I am not against this change because of unique constrains on the table it should work, just wondering.

Reply:
Any call to setPillar changes the pillar, see the example in PR description.